moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -577,51 +577,89 @@ class LoadPR2 : public testing::Test
   moveit::core::RobotModelConstPtr robot_model;
 };
 
-void generateTestTraj(std::vector<std::shared_ptr<robot_state::RobotState>>& traj,
-                      const moveit::core::RobotModelConstPtr& robot_model,
-                      const robot_model::JointModelGroup* joint_model_group)
+std::size_t generateTestTraj(std::vector<std::shared_ptr<robot_state::RobotState>>& traj,
```

Comment:
I don't understand why you made these tests so much more complicated. This doesn't add any functionality.
Please revert this part or explain.

Reply:
I did a few things here. For one I now use the PR2's default values as a base instead of a vector of all zeros since the zero vector is out of the joint limits for the PR2. I also changed the tests to use the testJointSpaceJump with the struct parameter rather than calling the absolute and relative tests directly. To make it clear which test is being used by testJointSpaceJump under the hood I changed the test trajectory so that the absolute jump threshold and relative jump threshold trip at different waypoints.

Reply:
I noticed these changes. But I think they are not necessary to perform the testing. Particularly adding many more waypoints make the whole test less comprehensible (for a short glance). Having exactly a single jump is much clearer.

Reply:
I reverted the test to 4 waypoints with a single obvious jump at the end